### PR TITLE
add calorimeter python module

### LINKED ISSUE #2760: no python module for particle calorimeter
There is currently no python module to automatically read out the particle calorimeter data. This should be added.

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request closes #2760 by adding a python module that allows automated data extraction from the particle calorimeter plugin.

### CODE REVIEW

**Comment by PrometheusPi** [comment]
fixed the pep8 issues reported by flake

File: lib/python/picongpu/extra/plugins/data/calorimeter.py
Comment (steindev) [inline lib/python/picongpu/extra/plugins/data/calorimeter.py:1]: Please specify the units of the returned values.

File: docs/source/usage/plugins/particleCalorimeter.rst [hunk 1]
```diff
@@ -154,19 +153,29 @@ Analysis Tools
...
+calObj = particleCalorimeter("./simOutput/e_calorimeter/e_calorimeter_all_%T.bp")
+
+# last bin contains overflow 
+selected_energy_bin = -1
+
+plt.title("selected energy: >{:.1f} keV".format(calObj.getEnergy()[selected_energy_bin]), fontsize=18)
+
+plt.pcolormesh(calObj.getYaw(), calObj.getPitch(), calObj.getData(2000)[selected_energy_bin, :, :])
+
+plt.xlabel(calObj.detector_params["axisLabels"][-1], fontsize=18)
```
Comment (steindev) [inline docs/source/usage/plugins/particleCalorimeter.rst]: Please check if units of the axis are specified.

Comment (PrometheusPi) [reply docs/source/usage/plugins/particleCalorimeter.rst]: Good catch - while the dict names contain the unit, the labels themselves do not contain the unit. I will provide a fix.

File: lib/python/picongpu/extra/plugins/data/calorimeter.py
Comment (PrometheusPi) [reply lib/python/picongpu/extra/plugins/data/calorimeter.py:1]: fixed

File: docs/source/usage/plugins/particleCalorimeter.rst (hunk 1 above)
Comment (PrometheusPi) [reply docs/source/usage/plugins/particleCalorimeter.rst]: fixed

File: docs/source/usage/plugins/particleCalorimeter.rst
Comment (steindev) [inline docs/source/usage/plugins/particleCalorimeter.rst:1]: I just saw in built documentation the python code block is not correctly rendered. Maybe you missed some '```' after `.. code:: python` or forgot to indent?

See (https://picongpu--4697.org.readthedocs.build/en/4697/usage/plugins/particleCalorimeter.html#analysis-tools)

File: docs/source/usage/plugins/particleCalorimeter.rst [hunk 2]
```diff
@@ -154,19 +153,29 @@ Analysis Tools
...
+
+plt.title("selected energy: >{:.1f} keV".format(calObj.getEnergy()[selected_energy_bin]), fontsize=18)
+
+plt.pcolormesh(calObj.getYaw(), calObj.getPitch(), calObj.getData(2000)[selected_energy_bin, :, :])
+
+plt.xlabel(calObj.detector_params["axisLabels"][-1] + r" $[^\circ]$", fontsize=18)
+plt.ylabel(calObj.detector_params["axisLabels"][-2] + r" $[^\circ]$", fontsize=18)
+
+cb = plt.colorbar()
+cb.set_label("energy", fontsize=18)
```
Comment (steindev) [inline docs/source/usage/plugins/particleCalorimeter.rst]: The unit of the energy axis is still missing.

**Review by steindev — CHANGES_REQUESTED** [review]

File: docs/source/usage/plugins/particleCalorimeter.rst
Comment (PrometheusPi) [reply docs/source/usage/plugins/particleCalorimeter.rst:1]: fixed - leading white spaces were missing

File: docs/source/usage/plugins/particleCalorimeter.rst (hunk 2 above)
Comment (PrometheusPi) [reply docs/source/usage/plugins/particleCalorimeter.rst]: thanks - fixed

**Comment by PrometheusPi** [comment]
@steindev everything has been fixed - you can re-review

**Comment by PrometheusPi** [comment]
ping @steindev

**Comment by PrometheusPi** [comment]
New month: ping @steindev

**Review by steindev — APPROVED** [review]

**Pull request merged**